src/main/java/dev/meyi/bn/modules/Module.java

```diff
@@ -12,14 +13,14 @@ public abstract class Module {
 
   protected int x;
   protected int y;
-  protected float scale;
+  public float scale;

```

Comment:
Instead of making this public, keep it protected and add a getter & setter for it and use those. It's protected for a reason. we don't want to break encapsulation like that.